titus_isolate/monitor/utils.py

```diff
@@ -55,3 +59,87 @@ def is_core_below_threshold(
         log.debug("Core: {} with usage: {} is OVER  threshold: {}".format(core.get_id(), usage, threshold))
 
     return is_free
+
+
+def normalize_monotonic_data(timestamps, buffers, num_buckets=60, bucket_size_secs=60) -> List[float]:
+    return __normalize_data(__get_monotonic_element, timestamps, buffers, num_buckets, bucket_size_secs)
+
+
+def normalize_gauge_data(timestamps, buffers, num_buckets=60, bucket_size_secs=60) -> List[float]:
+    return __normalize_data(__get_gauge_element, timestamps, buffers, num_buckets, bucket_size_secs)
+
+
+def __normalize_data(get_element, timestamps, buffers, num_buckets, bucket_size_secs) -> List[float]:
+    data = np.full((num_buckets,), np.nan, dtype=np.float32)
+    if len(timestamps) == 0:
+        return list(data.tolist())
+
+    min_indices, max_indices = __get_bucket_indices(timestamps, num_buckets, bucket_size_secs)
+
+    for i in range(len(min_indices)):
+        min_index = min_indices[i]
+        max_index = max_indices[i]
+
+        if min_index < 0 or max_index < 0:
+            continue
+
+        data[i] = get_element(timestamps, buffers, min_index, max_index)
+
+    return list(data.tolist())
+
+
+def __get_monotonic_element(timestamps, buffers, min_index, max_index) -> float:
+    time_diff_ns = (timestamps[max_index] - timestamps[min_index]) * 1000000000
+    s = 0.0
+    for b in buffers:
+        s += b[max_index] - b[min_index]
+    if time_diff_ns > 0:
+        s /= time_diff_ns
+
+    return s
+
+
+def __get_gauge_element(timestamps, buffers, min_index, max_index) -> float:
```

Comment:
@rostyboost : I'm averaging the gauge elements in the indicated time slice here.  Do you happen to know how Atlas is doing it?